src/objects.rs

```diff
@@ -28,15 +31,17 @@ impl ObjectGroup {
         attrs: Vec<OwnedAttribute>,
         layer_index: Option<u32>,
     ) -> Result<ObjectGroup, TiledError> {
-        let ((o, v, c, n), ()) = get_attrs!(
+        let ((o, v, c, n), id) = get_attrs!(
             attrs,
             optionals: [
                 ("opacity", opacity, |v:String| v.parse().ok()),
                 ("visible", visible, |v:String| v.parse().ok().map(|x:i32| x == 1)),
                 ("color", colour, |v:String| v.parse().ok()),
                 ("name", name, |v:String| v.into()),
             ],
-            required: [],
+            required: [
+                ("id", id, |v:String| v.parse().ok()),
+            ],
             TiledError::MalformedAttributes("object groups must have a name".to_string())
```

Comment:
If they must have a name, should `name` be moved to `required`? I don't see why they should have a name though, so maybe the message should be fixed.

Reply:
Yeah I think that message should change, you can get a layer to have no `name` attribute just by setting it to empty in Tiled.
Not sure why it's optional here but required in `Layer`, they should both be optional I think.